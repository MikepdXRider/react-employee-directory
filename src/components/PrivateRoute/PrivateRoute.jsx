// 
// PrivateRoute Component
// Needs access to user/auth context
// Import usehistory hook
// This functional component takes in children and ...rest
// It returns a route
//      - said route has a render attribute which will render children or redirect user to auth route depending on user/auth state.
// 