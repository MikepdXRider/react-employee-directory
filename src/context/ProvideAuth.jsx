// 
// UserContext
// import hook and createContext
// call createContext hook
// create UserProvider w/ initialized state
//  - pass state mutators and any relevant functions into provider value attribute
//  - export provider and context
// 