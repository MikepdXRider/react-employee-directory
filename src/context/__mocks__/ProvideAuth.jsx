import { createContext, useState } from "react";

const authContext = createContext();

function ProvideAuth({mockUser, children}) {
    // on app load...
    // getUser checks the 'session' in localstorage for a current token and a refresh token. If the token is current, it looks for a user in the session. If the token is not current and a refresh token is present, it refreshes the token and retrieves a new user.
    const [user, setUser] = useState(
        mockUser ? {email: mockUser.email} : {}
    );

    return (
        <authContext.Provider value={{user}} >
            {children}
        </authContext.Provider>
    )
}

export { authContext, ProvideAuth}