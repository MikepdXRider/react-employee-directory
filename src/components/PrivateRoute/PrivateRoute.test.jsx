import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import { ProvideAuth } from '../../context/ProvideAuth.jsx';
import PrivateRoute from './PrivateRoute.jsx';

jest.mock('../../context/ProvideAuth.jsx');

it('renders PrivateRoute', () => {
    const {container} = render(
        <ProvideAuth>
            <MemoryRouter>
                <PrivateRoute location='/'/>
            </MemoryRouter>
        </ProvideAuth>
    )

    expect(container).toMatchSnapshot();
})