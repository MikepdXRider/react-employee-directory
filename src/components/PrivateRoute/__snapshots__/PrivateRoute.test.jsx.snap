// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PrivateRoute 1`] = `<div />`;
